modules/mapper-extras/src/test/java/org/elasticsearch/index/mapper/extras/ScaledFloatFieldMapperTests.java

```diff
@@ -156,6 +156,22 @@ public void testNoDocValues() throws Exception {
         assertEquals(1230, pointField.numericValue().longValue());
     }
 
+    public void testDocValuesSearchable() throws Exception {
+        MapperService mapperService = createMapperService(fieldMapping(b -> b.field("type", "scaled_float").field("scaling_factor", 10.0)));
+        MappedFieldType fieldType = mapperService.fieldType("field");
+        assertTrue(fieldType.isSearchable());
+    }
+
+    public void testDocValuesUnsearchable() throws Exception {
+        MapperService mapperService = createMapperService(
+            fieldMapping(
+                b -> b.field("type", "scaled_float").field("index", false).field("doc_values", false).field("scaling_factor", 10.0)
+            )
+        );
+        MappedFieldType fieldType = mapperService.fieldType("field");
+        assertFalse(fieldType.isSearchable());
+    }
+
```

Comment:
Maybe also add unit tests for the following cases: 

- index=false, doc_values=true and assert that field is searchable.
- index=true, doc_values=false and assert that field is searchable.
- defaults and time_series_metric=gauge and assert that field is searchable.